### ISSUE TITLE
proto: move configuration of tracer provider implementations out of `config.trace` into `extensions.tracers.<provider>`

### ISSUE THREAD

**Issue by yskopets** [envoyproxy/envoy#10737]
*Title*: move configuration of tracer provider implementations out of `config.trace` into `extensions.tracers.<provider>`

*Context*:
* proper placement of extension-specific configuration would have prevented #10576

**Comment by mattklein123** [comment]
cc @sanjaypujare can you potentially do this as part of solving the OpenCencus issue? It should be straightforward.

**Comment by sanjaypujare** [comment]
See my comment in https://github.com/envoyproxy/envoy/issues/10576#issuecomment-612179006 . I can get it to it later but not now.

**Comment by yskopets** [comment]
@sanjaypujare  @mattklein123 I will take a look into it.

**Comment by yskopets** [comment]
@mattklein123 

Just to clarify, what do you think should be in the scope of this task ?

E.g.,
* I could  just add `udpa.annotations.file_migrate` to provider-specific message types and they will be moved into `extensions.tracers.<provider>` in `v4alpha`
* or, do you expect these types to be moved into `extensions.tracers.<provider>` already in `v3` ? I think, it's doable, but it requires more automation from `api/tools` to keep `config.trace.v3` and `extensions.tracers.<provider>.v3` in sync, so that users could configure Envoy with any of these types

WDYT ?

**Comment by mattklein123** [comment]
@yskopets I think just doing this for v4alpha is fine but I will defer to @htuch on the best plan forward. Thank you!

**Comment by htuch** [comment]
Yeah, I think this is best done in v4alpha to avoid breaking API changes, thanks.
